File: src/common/py/data/entities/user/user_settings.py
import typing
from dataclasses import dataclass, field

from dataclasses_json import dataclass_json

from ..connector import ConnectorInstance


@dataclass_json
@dataclass(frozen=True, kw_only=True)
class UserSettings:
    """
    User settings (i.e., the settings a user configures in the UI) data.

    Attributes:
        connector_instances: A list of all configured connector instances.
    """

    connector_instances: typing.List[ConnectorInstance] = field(default_factory=list)

File: src/common/web/data/entities/user/UserSettings.ts
import { Type } from "class-transformer";

import { ConnectorInstance } from "../connector/ConnectorInstance";

/**
 * User settings (i.e., the settings a user configures in the UI) data.
 *
 * @param connector_instances - A list of all configured connector instances.
 */
export class UserSettings {
    // @ts-ignore
    @Type(() => ConnectorInstance)
    public readonly connector_instances: ConnectorInstance[] = [];

    public constructor(connectorInstances: ConnectorInstance[] = []) {
        this.connector_instances = connectorInstances;
    }
}
